gsoc2018-bharat/src/WikiDetector.py

```diff
@@ -0,0 +1,167 @@
+import os
+import re
+import sys
+import time
+import tempfile
+from joblib import Parallel, delayed
+from urllib.parse import unquote
+from collections import Counter
+
+def upcase_first_letter(s):
+	"""
+	Capitalize the string.
+	"""
+    return s[0].upper() + s[1:]
+
+def replaceAnchorText(filename):
+	"""
+	Given the input file, the surface forms loaded from anchor text
+	are used to extract entity mentions and replace them with the
+	article title in the text corpus itself.
+
+	Arguments
+	---------
+	filename : Input file containing the extracted text.
+	"""
+	print(filename)
+
+	# A temporary file to track the input file document by document.
+	t = tempfile.NamedTemporaryFile(mode = "r+")
+	dictionary = {}
+	with open(filename, 'r') as fil:
```

Comment:
this code block is a bit too long. Might be better to isolate that into separate functions